shell/browser/native_window_mac.mm

```diff
@@ -1281,18 +1282,15 @@ void ViewDidMoveToSuperview(NSView* self, SEL _cmd) {
 void NativeWindowMac::SetVisibleOnAllWorkspaces(bool visible,
                                                 bool visibleOnFullScreen,
                                                 bool skipTransformProcessType) {
-  // In order for NSWindows to be visible on fullscreen we need to functionally
-  // mimic app.dock.hide() since Apple changed the underlying functionality of
+  // In order for NSWindows to be visible on fullscreen we need to invoke
+  // app.dock.hide() since Apple changed the underlying functionality of
   // NSWindows starting with 10.14 to disallow NSWindows from floating on top of
   // fullscreen apps.
   if (!skipTransformProcessType) {
-    ProcessSerialNumber psn = {0, kCurrentProcess};
     if (visibleOnFullScreen) {
-      [window_ setCanHide:NO];
-      TransformProcessType(&psn, kProcessTransformToUIElementApplication);
+      Browser::Get()->DockHide();
```

Comment:
:+1: on reusing the workarounds found in DockHide() / DockShow() here.

This diff has a subtle change: instead of calling `NSWindow setCanHide:NO]` for just `window_`, `Browser::DockHide()` calls it for everything in `WindowList::GetWindows()`. Is that OK / intentional here?

Reply:
i'd say yes - imo the previous version (only doing it on `window_`) was incorrect